esx_service/utils/vmdk_perf.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2016 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##
+## Implements the interface for side car based implementation
+## of a KV store for vmdks.
+##
+
+from ctypes import *
+from vmware import vsi
+import pyVim
+import pyVim
+from pyVim.connect import Connect, Disconnect
+from pyVim import vmconfig
+from pyVmomi import VmomiSupport, vim, vmodl
+import sys
+import logging
+import atexit
+
+# Format to return stats
+FORMAT_CSV = vim.PerfFormat.csv
+DEFAULT_PERF_FORMAT = FORMAT_CSV
+
+# IO stats returned for the volume
+IO_STATS = 'iostats'
+
+# Items per metric counter
+LABEL = 'label'
+UNIT_INFO = 'unit_info'
+SUMMARY = 'summary'
+NUM_TYPE = 'num'
+IGN_TYPE = 'Ignore'
+
+# Performance counters per vm:device
+COUNTERS='counters'
+
+# Default collection interval
+COLLECTION_INTERVAL = 20
+
+# Samples to retrieve
+SAMPLE_SIZE = 1
+DEFAULT_SAMPLE_SIZE = SAMPLE_SIZE
+
+# Global map of counter IDs supported by the performance manager.
+perf_counters_map = {}
+
+# Global map of VMs and devices and performance counter IDs
+vm_dev_map = {}
+
+# Performance manager from the service instance
+perfm = None
+
+# The local service instance used by this module
+si = None
+
+# Metric short names
+metric_short_labels = {
+   "Average number of outstanding read requests": 'avgInProgRds',
+   "Average number of outstanding write requests": 'avgInProgWrs',
+   "Average read requests per second": 'avgRd/s',
+   "Average write requests per second": 'avgWr/s',
+   "Number of large seeks": 'largeSeeks(>8192 lbns)',
+   "Number of medium seeks": 'medSeeks(64 - 8192 lbns)',
+   "Number of small seeks": 'smallSeeks(<64 lbns)',
+   "Read Latency (us)": 'rdLat',
+   "Read latency": 'avgRdLat',
+   "Read rate": 'volRdRate',
+   "Read request size": 'avgRdRqSz(bytes)',
+   "Read workload metric": 'Ignore',
+   "Write Latency (us)": 'wrLat',
+   "Write latency": 'avgWrLat',
+   "Write rate": 'volWrRate',
+   "Write request size": 'avgWrRqSz(bytes)',
+   "Write workload metric": 'Ignore'
+}
+
+
+def init_svc_inst():
+    '''
+    connect and do stuff on local machine
+    '''
+    global si
+    global perfm
+
+    # Connect to localhost as dcui
+    # User "dcui" is a local Admin that does not lose permissions
+    # when the host is in lockdown mode.
+    si = pyVim.connect.Connect(host='localhost', user='dcui')
+    if not si:
+        raise SystemExit("Failed to connect to localhost as 'dcui'.")
```

Comment:
Can it be a tmp issue ? I guess just skipping the perf collection for a cycle and trying again is a better reaction - what do you think ?